dubbo-registry/dubbo-registry-zookeeper/src/main/java/org/apache/dubbo/registry/zookeeper/ZookeeperRegistry.java

```diff
@@ -126,6 +151,41 @@ public void doUnregister(URL url) {
         }
     }
 
+    private void checkZkProvider(){
+        Set<URL> registeredURLSet = new HashSet<URL>(getRegistered());
+        for (URL registeredURL: registeredURLSet){
+            if(registeredURL.getProtocol().equalsIgnoreCase(Constants.CONSUMER))
+                continue;
+
+            String host = registeredURL.getHost();
+            int port = registeredURL.getPort();
+            String interfaceName = registeredURL.getServiceInterface();
+            List<String> childPath = zkClient.getChildren(Constants.PATH_SEPARATOR + DEFAULT_ROOT + Constants.PATH_SEPARATOR
+                    + interfaceName + Constants.PATH_SEPARATOR + Constants.PROVIDERS_CATEGORY);

```

Comment:
can be refactored to a new method

List<String> childPath = zkClient.getChildren(registeredURL.getServiceInterface());
private String getZKKey(String interfaceName) {
   return Constants.PATH_SEPARATOR + DEFAULT_ROOT + Constants.PATH_SEPARATOR
                    + interfaceName + Constants.PATH_SEPARATOR + Constants.PROVIDERS_CATEGORY;
}